### DISCUSSION TITLE
Localization on attributes

### DISCUSSION THREAD

**Discussion by heischo** [dotnet/aspnetcore#51355] [Ideas] [answered]
[StringLength(3, ErrorMessage = "Name is too long")]

Can we use localization in such attributes, in a future ASP.NET Core version, without writing our own attributes? 
I am thinking of a special keyword together with the localization name as a kind of placeholder, which is then filled with the correct value by the runtime:

[StringLength(3, ErrorMessage = localizationRef("TooLongErrorMessage"))]

**Comment by martincostello** [comment, marked as answer]
Have you tried the approaches documented here? https://learn.microsoft.com/aspnet/core/fundamentals/localization/make-content-localizable#dataannotations-localization

**Reply by heischo** [reply]
No. I will have a look at that. But I still think that having a separate keyword could make things a lot easier for us.

**Reply by martincostello** [reply]
A new language keyword would be a compiler feature, so would be outside the remit of the aspnetcore repo and would be for Roslyn.
